src/signals-unix.c

```diff
@@ -470,10 +474,11 @@ void usr2_handler(int sig, siginfo_t *info, void *ctx)
     if (ptls == NULL)
         return;
     int errno_save = errno;
-    sig_atomic_t request = jl_atomic_exchange(&ptls->signal_request, 0);
+    sig_atomic_t request = jl_atomic_exchange(&ptls->signal_request, -1);
 #if !defined(JL_DISABLE_LIBUNWIND)
     if (request == 1) {
         signal_context = jl_to_bt_context(ctx);
+        jl_atomic_exchange(&ptls->signal_request, 0);
 
         pthread_mutex_lock(&in_signal_lock);
         pthread_cond_broadcast(&signal_caught_cond);
```

Comment:
@staticfloat it was previously indeterminant whether we had passed this statement before `pthread_cond_timedwait` returned. Now we know (from the value of `signal_request` before we reset it to `0`) whether we had not yet reached line 481

Reply:
Thanks; but I meant for this to be encoded in the source code so that others can see it in the future as well.